%!TEX root = main.tex

\section*{Introduction}

Genetic clustering algorithms are a commonly used approach to describe genetic structure among populations, reflecting the cumulative effects of isolation, connectivity (i.e., genetic migration), selection, and genetic drift.  The ability of such algorithms to differentiate populations is limited by the extent of genetic differentiation among populations and the capacity to resolve those differences given the available marker set and sample sizes from the respective populations.  Based on an initial description of genetic structure, ecologists are frequently interested in describing patterns of genetic movement among populations, often inferred from the mismatch between the sampling location and genetic origins of a given animal \citep{paetkau1995microsatellite,wilson2003bayesian}.  Describing such patterns of movement have many implications, influencing evolutionary processes, disease dynamics \citep{huestis2019windborne}, or patterns of invasive species expansion.

	In using clustering algorithms to describe genetic structure among natural populations, researchers frequently identify individuals with complex ancestry patterns (i.e., the proportion of an individual’s genome assigned to sampled populations, sensu \citealt{pritchard2000inference}) that demonstrate mixed associations to multiple populations.  Various processes could contribute to such mixed associations.  Most simplistically, individuals may represent the influences of gene flow among populations, occurring in recent or prior generations, with the observed complex ancestry patterns accurately representing contributions from multiple populations included in the study.  However, alternative processes could also create similarly complex ancestry patterns.  For example, individuals may have genomic associations to unsampled populations as immigrants or as descendant of immigrant from unsampled populations \citep{pritchard2000inference,beerli2004effect}.  Yet a limitation of clustering algorithms is that exogenous immigrants can only be associated to populations represented within the sample.  Similarly, individuals with mixed associations may reflect a discrepancy between the genetic structure as described by the analysis and the underlying ecological process creating patterns of genetic differentiation (e.g., when clinal systems governed by isolation by distance are described as discrete genetic clusters).  Finally, such patterns may represent a limitation of the statistical power of the genetic data, as a function of both the marker set and sample size, to resolve the true underlying patterns of genetic structure.  The challenge for researchers then becomes correctly identifying the ecological process by which similar, complex ancestry patterns were created.  
	
Pairing empirical data with closely derived simulation-based approaches has been long recognized as a tool to help guide the interpretation of empirical data and define the limitations of the available data \citep{vaha2006efficiency,anderson2008improved,latch2011fine}.  Mirroring empirical datasets with forward-in-time simulations often implement sampling with replacement (e.g. citations).  However, sampling with replacement implicitly increases the pairwise genetic distance (i.e., FST) among populations and inflates the perceived ability to correctly assign individuals back to source populations \cite{almudevar2000exact,anderson2008improved}.  \textcolor{red}{Seems like 2 additional sentences here would be appropriate to draw from a study that has demonstrated this process.}  The effects of overestimating our ability to resolve populations creates individuals that discretely align with simulated populations.  These issues could be addressed with simulations that fix observed genetic differences among populations by sampling from the empirical data without replacement (citations for simulation frameworks that use this approach).
  Beyond simply evaluating the resolution of discrete populations in a simulated framework, researchers may also be interested in how individuals of admixed ancestry are genetically characterized.  Recognizing that connectivity among genetically distinct populations may be relatively rare, researchers can gain additional insights into the frequency of immigration by also identifying descendants of migrants among study populations (directly [F1 hybrids] or advanced hybrids reflecting the effect of backcrossing across multiple generations [BC1, \ldots, BCX]).  The process of recombination during meiosis creates a distribution of expected associations to parental populations among advanced hybrid classes. Treating linked loci as independent can lead to the false interpretation that the association of advanced hybrid classes to parental population occur strictly as discrete distributions.  Therefore, in evaluating descendants of migrants within a simulated framework, it is imperative to take into consideration the pattern of linkage among loci available for analysis.
	In response to these needs, we developed gscramble – an approach to simulate genotypes that minimizes bias associated with forward-in-time simulations by sampling alleles from respective populations without replacement, thus preserving the genetic differences observed in the empirical dataset.  Further, by simulating individual genotypes based on user-defined pedigrees, gscramble allows for the simulation of admixed genotypes of varying degrees of complexity while allowing the tracking of haplotype blocks from source populations within the specified pedigree.  By integrating species-specific recombination rates, gscramble simulates biologically informed genotypes that mirror empirical data.  We develop and illustrate gscramble with use of single nucleotide polymorphic (SNP) genotypes.



  

\section*{Methods}

\subsection*{An introductory motivating example}

Before we proceed to a description of the methodology used by gscramble,
we demonstrate a case in which using a standard sampling-with-replacement
approach to simulate genotypes leads to the incorrect conclusion that a set of genetic
markers is sufficient to accurately estimate admixture fractions between two populations.
To construct this simple example, we used the R programming language (CITE) to
simulate allele frequencies for $L$ biallelic
loci in a single population from a $\mathrm{Beta}(1, 8)$ distribution and then simulated
two samples, $A$ and $B$ of size $N$ diploid genotypes from this population by drawing each of the three
possible genotypes at each locus from their
expected genotype frequencies under Hardy-Weinberg equilibrium.  In this case,
each set of $N$ individuals is a sample from exactly the same population, so
there would be no basis for performing population assignment between $A$ and $B$.  

However, we treated samples $A$ and $B$ as if they were sampled from two potentially different
populations, and simulated $9n$ new individuals: $n$ new individuals at each of the 9 values of
$q_A$, the admixture fraction for population $A$, in  $\{0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1.0\}$.
We then added these $9n$ to the original $2N$ individuals ($N$ from sample $A$ and $N$ from $B$)
and analyzed the genotypes using ADMIXTURE with $K=2$, doing a completely
unsupervised analysis or doing a supervised analysis in which the original $N$ samples from each
population were provided as learning samples. 

For each combination $L \in \{10^2, 10^3, 10^4, 10^5\}$, 
$N \in \{25, 50, 100, 250\}$ and $n\in\{3,12,24\}$ we conducted multiple replicates
of the entire process of
simulating samples $A$ and $B$, simulating the $9n$ additional individuals, and
then analyzing them with ADMIXTURE.  We conducted sufficient replicates so that,
for each combination of $L$, $N$, and $n$ we had simulated 480 new individuals
at each $q_A$ value; thus, $160$ replicates when $n = 3$, $40$ when $n=12$, and
$20$ when $n=24$.

We hypothesized that ADMIXTURE would return estimates of $q_A$ that were
centered around the true values, even though there is no basis for that, because
the simulations were done in a na\"{i}ve way that effectively assumes that the population
allele frequencies are identical to those observed in the samples themselves. 
We further expected that ADMIXTURE would return more accurate estimates of
$q_A$ when $N$ was smaller and the number of loci was larger.
We summarized the results by plotting boxplots of the 480
ADMIXTURE-inferred $q_A$ values, for each combination of $L$, $N$, and $n$.




\subsection*{Genomic Simulation Pedigrees}

For the purposes here, we define blah, blah.

\subsection*{Another subsection}



\subsection*{And a third}

Blah, blah blobbity blah.


\begin{figure*}
\begin{center}
\includegraphics[width=0.8\textwidth]{images/gsp4-700.pdf}
\end{center}
\caption[]{Stick this thing in somewhere}
\end{figure*}


\section*{Results}

\subsection*{An introductory motivating example}

As expected, the perceived ability to estimate $q_A$ with ADMIXTURE was higher
for larger values of $L$ and for smaller values of $N$ (Figure~\ref{fig:bias-sims}).  
%%%
\begin{figure*}
\includegraphics[width=\textwidth]{figures/bias-sims-unsup-and-sup.pdf}
\caption{ADMIXTURE estimates of $q_A$ as described in
{\em An introductory motivating example} In these results, the ability to estimate $q_A$
is a product of a bias.
}
\label{fig:bias-sims}
\end{figure*}
%%%
It is interesting to note
that, for numbers of loci,  $L$, that are 100 (or smaller) the effect is observed only when
sample size, $N$, is as small as 25---and even then the effect is slight.  However, when the
number of markers increases to 100,000---values that are becoming commonplace with the availability of chip-
or sequencing-based approaches---it is apparent that the sampling-with-replacement approach
induces an extreme bias.  With 100,000 markers, even if sample sizes are as large as 250 individuals,
using na\"{i}ve simulations improperly indicate that admixture fractions of individuals can be
accurately estimated, even when there are no genetic differences, whatsoever between the populations that
samples $A$ and $B$ are drawn.   
 


\section*{Discussion}
Hey! Let's discuss this!



\section*{Acknowledgements}
We thank etc. etc.   Contribution number  mHAVEAGAS-003